sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2332,9 +2332,14 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
         case owg: SupportsOrderingWithinGroup if u.isDistinct =>
           throw QueryCompilationErrors.distinctInverseDistributionFunctionUnsupportedError(
             owg.prettyName)
-        case owg: SupportsOrderingWithinGroup if u.orderingWithinGroup.isEmpty =>
+        case owg: SupportsOrderingWithinGroup
+          if !owg.orderingFilled && u.orderingWithinGroup.isEmpty =>
           throw QueryCompilationErrors.inverseDistributionFunctionMissingWithinGroupError(
```

Comment:
To confirm, all the error cases here are tested, right?

Reply:
Yes.